unit_tests/test_scan_operation_thread.py

```diff
@@ -151,80 +153,70 @@ def test_negative_prometheus_validation_error(events, cluster):
                 ScanOperationThread(default_params)._run_next_operation()  # pylint: disable=protected-access
             all_events = get_event_log_file(events)
             assert "Severity.NORMAL" in all_events[0] and "period_type=begin" in all_events[0]
-            assert "Severity.ERROR" in all_events[1] and "period_type=end" in all_events[
+            severity = "Severity.ERROR"
+            if SkipPerIssues('https://github.com/scylladb/scylladb/issues/21578', TestConfig().tester_obj().params):
+                severity = "Severity.WARNING"
+            assert severity in all_events[1] and "period_type=end" in all_events[
                 1] and "Fullscan failed - 'mapreduce_service_requests_dispatched_to_other_nodes' was not triggered" in all_events[1]
 
 
-class ExecuteOperationTimedOutMockCqlConnectionPatient(MockCqlConnectionPatient):
+class OperationTimedOutConnection(MockCqlConnectionPatient):
     def execute(*args, **kwargs):
         # pylint: disable=unused-argument
         # pylint: disable=no-method-argument
         raise OperationTimedOut("timeout")
 
 
-class ExecuteAsyncOperationTimedOutMockCqlConnectionPatient(MockCqlConnectionPatient):
+class AsyncOperationTimedOutConnection(MockCqlConnectionPatient):
     def execute_async(*args, **kwargs):
         # pylint: disable=unused-argument
         # pylint: disable=no-method-argument
         raise OperationTimedOut("timeout")
 
 
-@pytest.mark.parametrize(("mode", 'severity', 'timeout', 'execute_mock'),
-                         [['partition', 'WARNING', 0, 'execute_async'],
-                          ['aggregate', 'ERROR', 60*30, 'execute'],
-                          ['table', 'WARNING', 0, 'execute']])
-def test_scan_negative_operation_timed_out(mode, severity, timeout, execute_mock, events, node):
-    # pylint: disable=redefined-outer-name
-    # pylint: disable=too-many-arguments
-    if execute_mock == 'execute_async':
-        connection = ExecuteAsyncOperationTimedOutMockCqlConnectionPatient()
-    else:
-        connection = ExecuteOperationTimedOutMockCqlConnectionPatient()
-    db_cluster = DBCluster(connection, [node], {})
-    node.parent_cluster = db_cluster
-    default_params = ThreadParams(
-        db_cluster=db_cluster,
-        ks_cf='a.b',
-        mode=mode,
-        full_scan_aggregates_operation_limit=timeout,
-        full_scan_operation_limit=timeout,
-        **DEFAULT_PARAMS
-    )
-    with events.wait_for_n_events(events.get_events_logger(), count=2, timeout=10):
-        ScanOperationThread(default_params)._run_next_operation()  # pylint: disable=protected-access
-    all_events = get_event_log_file(events)
-    assert "Severity.NORMAL" in all_events[0] and "period_type=begin" in all_events[0]
-    assert f"Severity.{severity}" in all_events[1] and "period_type=end" in all_events[1]
+class ReadTimeoutConnection(MockCqlConnectionPatient):
```

Comment:
Could we use one class and pass exception as a input parameter instead? We could get rid of all of these duplicate classes and it would make parametrization easier as well.

I know this is not point of this PR, but since you change the class drastically anyway I would consider this as well

Reply:
didn't notice this optimization, thanks